fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/item/v1/SoundPlayer.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.item.v1;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.world.World;
+
+@FunctionalInterface
+public interface SoundPlayer {
```

Comment:
I do not like the name of this interface. To me it sounds like some sound related interface implemented onto the player.

To me it seems like a sort of preconfigured `SoundEffect` that you just pass the world and player listening to the sound into.